lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -87,126 +88,100 @@ bool connectShaderToMaterial(
-        return {};
+        return false;
     }
     if (UsdGeomScope(usdPrim) && usdPrim.GetName() == kDefaultMaterialScopeName) {
-        return {};
+        return false;
+    }
+    if (auto subset = UsdGeomSubset(usdPrim)) {
+        TfToken elementType;
+        subset.GetElementTypeAttr().Get(&elementType);
+        if (elementType != UsdGeomTokens->face) {
+            return false;
+        }
     }
     if (PXR_NS::UsdShadeMaterialBindingAPI::CanApply(usdPrim)) {
-        return usdPrim;
+        return true;
     }
-    return {};
+    return false;
 }
 
 BindMaterialUndoableCommand::BindMaterialUndoableCommand(
-    const UsdPrim& prim,
+    Ufe::Path      primPath,
     const SdfPath& materialPath)
-    : _stage(prim.GetStage())
-    , _primPath(prim.GetPath())
+    : _primPath(std::move(primPath))
     , _materialPath(materialPath)
 {
 }
 
 BindMaterialUndoableCommand::~BindMaterialUndoableCommand() { }
 
-void BindMaterialUndoableCommand::undo()
+void BindMaterialUndoableCommand::undo() { _undoableItem.undo(); }
+
+void BindMaterialUndoableCommand::redo() { _undoableItem.redo(); }
+
+void BindMaterialUndoableCommand::execute()
 {
-    if (!_stage || _primPath.IsEmpty() || _materialPath.IsEmpty()) {
+    if (_primPath.empty() || _materialPath.IsEmpty()) {
         return;
     }
 
-    UsdPrim prim = _stage->GetPrimAtPath(_primPath);
-    if (prim.IsValid()) {
-        auto bindingAPI = UsdShadeMaterialBindingAPI(prim);
-        if (bindingAPI) {
-            if (_previousMaterialPath.IsEmpty()) {
-                bindingAPI.UnbindDirectBinding();
-            } else {
-                UsdShadeMaterial material(_stage->GetPrimAtPath(_previousMaterialPath));
-                bindingAPI.Bind(material);
-            }
-        }
-        if (_appliedBindingAPI) {
-            prim.RemoveAPI<UsdShadeMaterialBindingAPI>();
-        }
+    UsdUndoBlock undoBlock(&_undoableItem);
+
+    auto prim = ufePathToPrim(_primPath);
+    if (!prim.IsValid()) {
+        return;
     }
-}
 
-void BindMaterialUndoableCommand::redo()
-{
-    if (!_stage || _primPath.IsEmpty() || _materialPath.IsEmpty()) {
+    UsdShadeMaterial material(prim.GetStage()->GetPrimAtPath(_materialPath));
+    if (!material) {
         return;
     }
 
-    UsdPrim          prim = _stage->GetPrimAtPath(_primPath);
-    UsdShadeMaterial material(_stage->GetPrimAtPath(_materialPath));
-    if (prim.IsValid() && material) {
-        UsdShadeMaterialBindingAPI bindingAPI;
-        if (prim.HasAPI<UsdShadeMaterialBindingAPI>()) {
-            bindingAPI = UsdShadeMaterialBindingAPI(prim);
-            _previousMaterialPath = bindingAPI.GetDirectBinding().GetMaterialPath();
-        } else {
-            bindingAPI = UsdShadeMaterialBindingAPI::Apply(prim);
-            _appliedBindingAPI = true;
-        }
-        bindingAPI.Bind(material);
+    if (auto subset = UsdGeomSubset(prim)) {
+        subset.GetFamilyNameAttr().Set(UsdShadeTokens->materialBind);
```

Comment:
That was the only required change to make assignment work and have the material display in the viewport.

Reply:
OK, but as I asked, IDk what are the implication of changing the family name on an existing subset.